src/Microsoft.ApplicationInsights.AspNetCore/Implementation/TelemetryConfigurationOptionsSetup.cs

```diff
@@ -115,6 +118,9 @@ public void Configure(TelemetryConfiguration configuration)
                     module.Initialize(configuration);
                 }
             }
+
+            // Microsoft.ApplicationInsights.DependencyCollector.DependencyTrackingTelemetryModule depends on this nullable configuration to support Correlation. 
+            configuration.ApplicationIdProvider = this.applicationIdProvider;
```

Comment:
@cijothomas Can you please confirm that this is the correct way to do this?

Reply:
Yes